lib/browser/guest-view-manager.js

```diff
@@ -184,6 +184,32 @@ const attachGuest = function (embedder, elementInstanceId, guestInstanceId, para
     disableBlinkFeatures: params.disableblinkfeatures
   }
 
+  // parse the 'webpreferences' attribute string, if set
+  // this uses the same parsing rules as window.open uses for its features
+  if (typeof params.webpreferences === 'string') {
+    // split the attribute's value by ','
+    let i, len
+    let webpreferencesTokens = params.webpreferences.split(/,\s*/)
```

Comment:
The code from `window.open` that parses this is left-over from when Electron was using `CoffeeScript` so it might be nicer to write this new code using ES2015, such as:

```js
params.webpreferences.split(/,\s*/).forEach((pref) => {
      let [name, value] = pref.split(/\s*=/)
      value = (value === 'yes' || value === '1') ? true : (value === 'no' || value === '0') ? false : value
      if (value === undefined) {
        value = true
      }
      webPreferences[name] = value
    })
```

Reply:
i can not set webpreferences to a object